Qin iotdir comments (#177)
anima-wg/brski-cloud: anima-brski-cloud.md

This kind of non-network onboarding is sometimes called "Application Onboarding", as the purpose is typically to deploy a credential that will be used by the device in its intended use.
For instance, a SIP phone might have a client certificate to be used with a SIP proxy.

> This document further specifies use of a BRSKI Cloud Registrar and clarifies operations that ere left out of scope in {{BRSKI}}.

Reviewer: ere -> are

> This document further specifies use of a BRSKI Cloud Registrar and clarifies operations that ere left out of scope in {{BRSKI}}.

Author: ```suggestion
This document further specifies use of a BRSKI Cloud Registrar and clarifies operations that are left out of scope in {{BRSKI}}.
```

According to {{BRSKI, Section 2.7}}, the Pledge MUST use an Implicit Trust Anchor database (see EST {{!RFC7030}}) to authenticate the Cloud Registrar service.
The Pledge MUST establish a mutually authenticated TLS connection with the Cloud Registrar.
Unlike the Provisional TLS procedures documented in {{BRSKI, Section 5.1}}, the Pledge MUST NOT establish a Provisional TLS connection with the Cloud Registrar.

> Pledges MUST and Cloud/Owner Registrars SHOULD support the use of the "server\_name" TLS extension (SNI, [RFC6066]) when using TLS 1.2.

Author: The Registrar can't use the SNI when the Pledge put the wrong one in.  So the Registrar basically has to ignore it.
This means only one Registrar per IP address/port combination, just like in the days before SNI.
That occurs whenever the Registrar connection is via a Join Proxy.  
It might also be the result of a port-forward from an firewall/edge-router to the Registrar.

Support for SNI is mandatory with TLS 1.3.

Pledges SHOULD send a valid "server\_name" extension (SNI) whenever they know the domain name of the registrar they connect to.
A Pledge creating a Provisional TLS connection according to {{BRSKI}} will often only know the IPv6 link local IP address of a Join Proxy that connects it to the Registrar.

> Registrars are according expected to ignore SNI information, as in most cases, the Pledge will not know how to set the SNI correctly.

Reviewer: according -> accordingly?

> Registrars are according expected to ignore SNI information, as in most cases, the Pledge will not know how to set the SNI correctly.

Author: ```suggestion
Registrars are accordingly expected to ignore SNI information, as in most cases, the Pledge will not know how to set the SNI correctly.
```

The Cloud Registrar must determine Pledge ownership.
Prior to ownership determination, the Registrar checks the request for correctness and if it is unwilling or unable to handle the request, it MUST return a suitable 4xx or 5xx error response to the Pledge as defined by {{BRSKI}} and HTTP.

> In the case of an unknown Pledge, a 404 is returned, for a malformed request, 400 is returned,

Reviewer: I had to read this few times to understand what this is trying to say.

How about something along the lines of the following instead:

The Registrar return the following errors:
* Unknown Pledge, return a 404
* Malformed Request, return 400
* Server Overload, return 503

> In the case of an unknown Pledge, a 404 is returned, for a malformed request, 400 is returned,
> or in case of server overload, 503 is returned.

Author: ```suggestion
The Registrar returns the following errors:

* in the case of an unknown Pledge, a 404 is returned, 
* for a malformed request, 400 is returned
* in case of server overload, 503 is returned.
```

## Considerations for HTTP Redirect {#considerationsfor-http-redirect}

When the default Cloud Registrar redirects a Pledge using HTTP 307 to an Owner Registrar, or another Cloud Registrar operated by a VAR, the Pledge MUST establish a Provisional TLS connection with the Registrar as specified in {{BRSKI}}.

> The Pledge is will be unable to determine whether it has been redirected to another Cloud Registrar that is operated by a VAR, or if it has been redirected to an Owner Registrar at this stage.

Reviewer: "Pledge is will be unable"
drop the "is"

> The Pledge is will be unable to determine whether it has been redirected to another Cloud Registrar that is operated by a VAR, or if it has been redirected to an Owner Registrar at this stage.

Author: ```suggestion
The Pledge will be unable to determine whether it has been redirected to another Cloud Registrar that is operated by a VAR, or if it has been redirected to an Owner Registrar at this stage.
```
